src/components/AddView.js

```diff
@@ -1,10 +1,78 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { db } from '../lib/firebase';
 
 const AddView = () => {
+  const [state, setState] = useState({
+    purchaseItem: '',
+    howSoon: '7',
+    lastPurchaseDate: null,
+  });
+
+  const handleChange = (e) => {
+    setState({
+      ...state,
+      [e.target.name]: e.target.value,
+    });
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (state.purchaseItem.trim()) {
+      let token =
+        localStorage.length !== 0 ? localStorage.getItem('token') : 'LA';
+      db.collection(token)
+        .add({
+          purchaseItem: state.purchaseItem,
+          howSoon: state.howSoon,
+          lastPurchaseDate: state.lastPurchaseDate,
+        })
+        .then(() => {
+          alert('Purchase Item has been recorded');
+        })
+        .catch((error) => {
+          alert(error.message);
+        });
+
+      setState({
+        purchaseItem: '',
+        howSoon: '7',
+        lastPurchaseDate: null,
+      });
+    } else {
+      alert('Please write an item');
+    }
+  };
+
   return (
-    <main>
-      <h1>Add Item</h1>
-    </main>
+    <>
+      <form className="form-container" onSubmit={handleSubmit}>
+        <label htmlFor="purchaseItem">Purchase Item Name:</label>
+        <input
```

Comment:
The input elements ought to have an `id` that corresponds with their label's `for` attribute.

Reply:
Yep. The semantics to tie the label to the input for users of assistive technology isn’t complete. All you need is to add an  id of  ```purchaseItem```
Also, to prepare for future styling,  it’d be good to wrap the label and input in a ```<div>```

Reply:
okay, thank you for the suggestion